From 36846c3f07a616790575bdc2678209ee30e39eaa Mon Sep 17 00:00:00 2001
From: raphaelchang <raphaelchang1227@gmail.com>
Date: Mon, 5 Nov 2018 05:11:33 -0500
Subject: [PATCH] Disable lcm-gen

---
 CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6d3a4c2..e71ecb0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,10 +30,9 @@ endif()
 
 # Core modules
 add_subdirectory(lcm)
-add_subdirectory(lcmgen)
 add_subdirectory(lcm-logger)
 
-option(LCM_ENABLE_EXAMPLES "Build test and example programs" ON)
+option(LCM_ENABLE_EXAMPLES "Build test and example programs" OFF)
 if(LCM_ENABLE_EXAMPLES)
   add_subdirectory(liblcm-test)
 endif()
@@ -79,7 +78,7 @@ lcm_option(
   "Build Go utilities, bindings is source distributed"
   GO_FOUND Go)
 
-option(LCM_ENABLE_TESTS "Build unit tests" ON)
+option(LCM_ENABLE_TESTS "Build unit tests" OFF)
 if(LCM_ENABLE_TESTS)
   enable_testing()
   add_subdirectory(test)
-- 
2.7.4

